powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/fido2/PowerAuthCryptographyService.java

```diff
@@ -67,13 +68,20 @@ public PowerAuthCryptographyService(ActivationRepository activationRepository, A
         this.certificateValidator = certificateValidator;
     }
 
-    public boolean verifySignatureForAssertion(String applicationId, String credentialId, CollectedClientData clientDataJSON, AuthenticatorData authData, byte[] signature, AuthenticatorDetail authenticatorDetail) throws GenericCryptoException, InvalidKeySpecException, CryptoProviderException, InvalidKeyException {
+    public boolean verifySignatureForAssertion(String applicationId, String credentialId, CollectedClientData clientDataJSON, AuthenticatorData authData, boolean isVisualSignature, byte[] signature, AuthenticatorDetail authenticatorDetail) throws GenericCryptoException, InvalidKeySpecException, CryptoProviderException, InvalidKeyException {
         if (!checkAndPersistCounter(applicationId, credentialId, authData.getSignCount())) {
             return false;
         }
+        byte[] dataSuffix = null;
+        if (isVisualSignature) {
+            dataSuffix = getOperationDataBytes(clientDataJSON.getChallenge());
+            if (dataSuffix == null) {
+                return false; // visual challenge expected, but no data found
```

Comment:
May we change this comment into DEBUG log message?

Reply:
Sounds good